### PR TITLE
chore: adr reconcile

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

Reconcile ADR 

## Related Issue

Fixes #

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/pepr/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: adr/0007-reconcile-action.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Date: 2024-02-08
+
+## Status
+
+---
+
+## Context
+
+As our `Watch` processor was enhanced to become more reliable, Pepr became a stronger candidate for Kubernetes Operator and Controller development. We needed a built in mechanism to guarantee state consistency of Kubernetes resources.
```
Comment (btlghrants) [inline adr/0007-reconcile-action.md]:
```suggestion
As our `Watch` processor was enhanced to become more reliable, Pepr became a stronger candidate for Kubernetes Operator and Controller development.  To facilitate this use case, we needed a built in mechanism for Pepr to guarantee consistent, ordered state changes for Kubernetes resources. 
```

File: adr/0007-reconcile-action.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+## Context
+
+As our `Watch` processor was enhanced to become more reliable, Pepr became a stronger candidate for Kubernetes Operator and Controller development. We needed a built in mechanism to guarantee state consistency of Kubernetes resources. 
+
+
+## Decision
+
+Add a `Queue` to the Watch controller as a new action called `Reoncile`. In some scenarios involving Kubernetes Resource Controllers or Operator patterns, utilizing a queue makes sense to sequentially handle events once they are returned by the Kubernetes API. This allows the queue to handle bursts of events without overwhelming the system or the Kubernetes API. It provides a mechanism to back off when the system is under heavy load, enhancing overall stability and maintaining the state consistency of Kubernetes resources, as the order of operations can impact the final state of a resource.
```
Comment (btlghrants) [inline adr/0007-reconcile-action.md]:
```suggestion
Add a `Queue` to the Watch controller and expose it as a new action called `Reconcile`. In many scenarios involving Kubernetes Resource Controller or Operator patterns, utilizing a queue makes sense to sequentially handle events as they are returned by the Kubernetes API.  It also allows handling of bursts of activity without overwhelming the system or the Kubernetes API. In addition, it provides a mechanism to back off when the system is under heavy load, enhancing overall stability and maintaining the state consistency of Kubernetes resources, as there are many cases in which the order of operations can impact the final state of a resource.
```

**Review by btlghrants — DISMISSED** [review]
Seems legit -- minor edits.

File: adr/0007-reconcile-action.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# 7: Reconcile Action
+
+Date: 2024-02-08
+
+## Status
+
+---
```
Comment (cmwylie19) [inline adr/0007-reconcile-action.md]:
```suggestion
Accepted
```

**Review by btlghrants — APPROVED** [review]
LGTM

**Pull request merged**